inc/parserutils.php

```diff
@@ -735,6 +735,62 @@ function p_get_first_heading($id, $render=METADATA_RENDER_USING_SIMPLE_CACHE){
 }
 
 /**
+ * Helper function for p_xhtml_cached_geshi:
+ * Parses additional options from options array and applies them to the
+ * given geshi instance.
+ * 
+ * The actually supported options are:
+ * - enable_line_numbers or enable_line_numbers=[GESHI_NORMAL_LINE_NUMBERS|GESHI_NO_LINE_NUMBERS]
+ * - start_line_numbers_at=integer
+ * - highlight_lines_extra="integer1,integer2..."
+ *
+ * @param  GeSHi    $geshi      Geshi instance/object
+ * @param  Array    $options    Options array
+ *
+ * @author LarsDW223
+ */
+function p_parse_geshi_options(&$geshi, array $options) {
```

Comment:
I don't think this function should exist. If there is option mangling to do, it should be done in parse_highlight_options() above. The result should be an option array that can be immeadiately used in p_xhtml_cached_geshi() below (eg, has all the right types etc.)